velox/functions/sparksql/tests/DecimalRoundTest.cpp

```diff
@@ -0,0 +1,164 @@
+/*
+ * Copyright (c) Facebook, Inc. and its affiliates.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include "velox/functions/sparksql/tests/SparkFunctionBaseTest.h"
+
+namespace facebook::velox::functions::sparksql::test {
+namespace {
+
+class DecimalRoundTest : public SparkFunctionBaseTest {
+ protected:
+  // For decimal round function, use aligned logic with Spark to compute the
+  // result type.
+  TypePtr getResultType(const TypePtr& type, int32_t roundScale) {
+    auto [precision, scale] = getDecimalPrecisionScale(*type);
+    int32_t integralLeastNumDigits = precision - scale + 1;
+    if (roundScale < 0) {
+      auto newPrecision = std::max(integralLeastNumDigits, -roundScale + 1);
+      return DECIMAL(std::min(newPrecision, 38), 0);
+    } else {
+      int32_t newScale = std::min(scale, roundScale);
+      return DECIMAL(std::min(integralLeastNumDigits + newScale, 38), newScale);
+    }
+  }
+
+  void testDictionary(
+      const core::CallTypedExprPtr& expr,
+      const std::vector<VectorPtr>& inputs,
+      const VectorPtr& expected) {
+    // Dictionary vectors as arguments.
+    auto newSize = inputs[0]->size() * 2;
+    std::vector<VectorPtr> vectors;
+    vectors.reserve(inputs.size());
+    for (auto i = 0; i < inputs.size(); ++i) {
+      auto indices = makeIndices(newSize, [&](int row) { return row / 2; });
+      vectors.emplace_back(
+          VectorTestBase::wrapInDictionary(indices, newSize, inputs[i]));
+    }
+    auto resultIndices = makeIndices(newSize, [&](int row) { return row / 2; });
+    auto expectedDic =
+        VectorTestBase::wrapInDictionary(resultIndices, newSize, expected);
+    auto actual = evaluate(expr, makeRowVector(vectors));
+    velox::test::assertEqualVectors(expectedDic, actual);
+  }
+
+  void testDecimalRound(
+      int32_t scale,
+      const VectorPtr& input,
+      const VectorPtr& expected) {
+    auto inputType = input->type();
+    std::vector<core::TypedExprPtr> inputs = {
+        std::make_shared<core::FieldAccessTypedExpr>(inputType, "c0")};
+    if (scale != 0) {
```

Comment:
It would be nice to allow testing decimal_round with zero scale specified explicitly as well. Consider, changing scale parameter type to std::optional, then pass it to the expression if set.